locals.tf

```diff
@@ -10,8 +20,9 @@ locals {
     (contains(["rapid", "stable", "node-image"], var.automatic_channel_upgrade) && var.kubernetes_version == null && var.orchestrator_version == null)
   )
   # Abstract the decision whether to create an Analytics Workspace or not.
-  create_analytics_solution  = var.log_analytics_workspace_enabled && var.log_analytics_solution == null
-  create_analytics_workspace = var.log_analytics_workspace_enabled && var.log_analytics_workspace == null
+  create_analytics_solution                       = var.log_analytics_workspace_enabled && var.log_analytics_solution == null
+  create_analytics_workspace                      = var.log_analytics_workspace_enabled && var.log_analytics_workspace == null
+  create_role_assignments_for_application_gateway = try(var.application_gateway_for_ingress.create_role_assignments, false)
```

Comment:
The role assignments should be created if `var.ingress_application_gateway_enabled = true`